crates/nu-parser/src/parser.rs

```diff
@@ -950,6 +950,9 @@ pub fn parse_internal_call(
 
         // Parse a positional arg if there is one
         if let Some(positional) = signature.get_positional(positional_idx) {
+            if positional.name == "def_name" && positional.var_id.is_none() {
```

Comment:
please do not check for a magic parameter name. This is considered bad coding practice.

Reply:
Is there any way to detect it's parsing def ? Ever we add some new parameter on parse_def function. 
Is there any way ?

Reply:
we can check it only with var_id too.
new tests added too .